ansible/roles/images/tasks/check-or-pull-images.yaml

```
@@ -0,0 +1,21 @@
+---
+# If image is present, its name is written to stdout. Return code is 0.
+#   > ctr --namespace k8s.io images check -q name==docker.io/mesosphere/pause-alpine:3.2; echo $?
+#   docker.io/mesosphere/pause-alpine:3.2
+#   0
+#
+# If image is not present, or is present, but some layer is missing, its name is not written to stdout. Return code is 0.
+#   > ctr --namespace k8s.io images check -q name==docker.io/mesosphere/pause-alpine:999; echo $?
+#   0
+- name: check existing image to ensure all content is available locally  # noqa no-changed-when
+  command: ctr --address {{ containerd_cri_socket }} --namespace k8s.io images check name=={{ image_name }}
```

Don't we want this (and other references in this PR)?
```suggestion
  command: ctr --address {{ flatcar_containerd_cri_socket if ansible_os_family == "Flatcar" else containerd_cri_socket }} --namespace k8s.io images check name=={{ image_name }}
```

---

I might be misunderstanding the provenance of this variable, but:
- It's defined in  https://github.com/mesosphere/konvoy-image-builder/blob/ae507387b4377294e9c1289e42917b32de5c0b5b/ansible/group_vars/all/defaults.yaml#L25
- And re-defined for flatcar in https://github.com/mesosphere/konvoy-image-builder/blob/125ae41a2940350c88b00698f6dfdaaa2930e3f9/ansible/vars/flatcar/flatcar.yaml#L6
- The re-definition is applied here: https://github.com/mesosphere/konvoy-image-builder/blob/e323bdcbdf3b9ed3b90fc00edfafb978e70f2e43/ansible/provision.yaml#L26-L30

If the above is right, then we should probably fix https://github.com/mesosphere/konvoy-image-builder/blob/504e73a2dad153e8d3c1ef6bb97930bcd7502fa2/ansible/roles/config/templates/crictl.yaml#L2,
https://github.com/mesosphere/konvoy-image-builder/blob/504e73a2dad153e8d3c1ef6bb97930bcd7502fa2/ansible/roles/config/templates/kubelet-drop-in.conf#L2,
and https://github.com/mesosphere/konvoy-image-builder/blob/a6aad4ebda80efe425fddf758b5db45fc246e43f/ansible/roles/packages/templates/config.toml.tmpl#L19-L23

---

Turns out I was wrong. After trying to figure it out with the help of https://docs.ansible.com/ansible/2.10/user_guide/playbooks_variables.html#variable-precedence-where-should-i-put-a-variable, I just gave up and put the if/else block everywhere.

This makes me wonder whether this code has any effect at all: https://github.com/mesosphere/konvoy-image-builder/blob/e323bdcbdf3b9ed3b90fc00edfafb978e70f2e43/ansible/provision.yaml#L26-L30